tensorflow/core/framework/tensor.h

```diff
@@ -35,6 +35,12 @@ limitations under the License.
 #include "tensorflow/core/platform/mem.h"
 #include "tensorflow/core/platform/types.h"
 
+#if !defined(PLUGGABLE_DEVICE_SUPPORTED) &&                                \
+    (__x86_64__ || __i386__ || defined(__APPLE__)) && !defined(ANDROID) && \
+    !defined(__ANDROID__) && !TARGET_OS_IOS
```

Comment:
These conditions are also for non-MacOS (any Linux on x86_64 and i386 will also satisfy the condition).
https://github.com/tensorflow/tensorflow/blob/5f7dcec3fabfa7b119790f87db20775088f0e2b7/tensorflow/core/kernels/list_kernels.h#L42-L49

To make it MacOS-specific, I think we only need
```suggestion
    defined(__APPLE__)) && !TARGET_OS_IOS
```

If you think this should apply to all PluggableDevice, then maybe we should rename it to something more generic.

Reply:
Done.